pkg/model/datacatalog/api.go

```diff
@@ -19,3 +19,20 @@ type GetAssetResponse struct {
 	// This value is assumed to be known to the catalog connector.
 	Credentials string `json:"credentials"`
 }
+
+// +fybrik:validation:object
+type CreateAssetRequest struct {
```

Comment:
Thanks @roytman for pointing out this issue. I have added the relevant descriptions. Please check the updated documentation and let me know if it is ok. 

However as I was  checking the current documentation I figured out other 2 minor issues which are described below. 

(1) There is  a minor inconsistency with the Details field as given [here](https://github.com/fybrik/fybrik/blob/052c0b7db24eeff77e387709fd305da2d9fad86e/pkg/model/datacatalog/api.go#L17) . As you can check in the previous link, the name of the field (which is `Details`) not matching with the go struct name  (which is `ResourceDetails`).  However, this inconsistency is not there if you check the `ResourceMetadata` field [here](https://github.com/fybrik/fybrik/blob/052c0b7db24eeff77e387709fd305da2d9fad86e/pkg/model/datacatalog/api.go#L16). This inconsistency has minor implications in the generation of mkdocs documentation. If you check the documentation, the link for the `Details` field is broken. 

So I thought it is better to change the field name to `ResourceDetails` from `Details` as [is done](https://github.com/fybrik/fybrik/blob/052c0b7db24eeff77e387709fd305da2d9fad86e/pkg/model/datacatalog/api.go#L16) for `ResourceMetadata` field. This leads to minor changes in other files in fybrik related to manager. Please review. 

(2) In the CreateAssetRequest go struct, I could not understand the real role of credentials field. It could be used to send the credentials of the destination data through Kubernetes secret. Hence I have made the field optional. Please review. Thanks. 

@shlomitk1 Please review from your side as well. Thanks.